katdal/test/test_applycal.py

```diff
@@ -456,11 +472,38 @@ def test_indirect_cal_product(self):
         self.test_gain_sensors('my_cal')
 
 
+class TestCalibrateFlux(object):
+    """Test :func:`~katdal.applycal.calibrate_flux` function."""
+
+    def setup(self):
+        gains = create_product(create_gain)
+        self.sensor = create_categorical_sensor(GAIN_EVENTS, gains, INVALID_GAIN)
+        calibrated_gains = create_product(partial(create_gain, fluxes=True))
+        self.calibrated_sensor = create_categorical_sensor(
+            GAIN_EVENTS, calibrated_gains, INVALID_GAIN)
+
+    def test_basic(self):
+        calibrated_sensor = calibrate_flux(self.sensor, TARGET_SENSOR, FLUXES)
+        assert_categorical_data_equal(calibrated_sensor, self.calibrated_sensor)
+
+    def test_missing_fluxes(self):
```

Comment:
Perhaps some further tests here testing user supplied gains. E.g. what if there is a 'measured_flux' target that isn't in the gain calibrator list?

Reply:
Easy to add.